lib/connection/commands.js

```diff
@@ -485,6 +459,9 @@ Response.prototype.parse = function(options) {
     promoteBuffers: promoteBuffers
   };
 
+  // Position within the message body at which documents start
+  this.index = 20;
```

Comment:
nooow I know why this was bothering me 😄 

So, you pre-parse the header and pass in the msgHeader and msgBody to this class. So you don't need to initialize the `this.index = 20`, because you really should be setting `this.data = msgBody` and keeping the index at 0. `this.raw` has to equal `msgHeader + msgBody`.